test/functional/tests/fault_injection/test_fault_injection_opencas_load.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm, casadm_parser, cli
+from api.cas.cache_config import CacheMode
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools import fs_utils
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem
+from test_utils.filesystem.file import File
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+test_file_path = f"{mount_point}/test_file"
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stop_no_flush_load_cache(cache_mode, filesystem):
+    """
+        title: Test to check that 'stop --no-data-flush' command works correctly.
+        description: |
+          Negative test of the ability of the CAS to load unflushed cache
+          when core device is mounted and unmounted.
+        pass_criteria:
+          - No system crash while load cache.
+          - Starting cache without loading metadata fails.
+          - Starting cache with loading metadata finishes with success.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_part = core_dev.partitions[0]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step(f"Add core with {filesystem.name} filesystem to cache and mount it."):
+        core_part.create_filesystem(filesystem)
+        core = cache.add_core(core_part)
+        core.mount(mount_point)
+
+    with TestRun.step(f"Create test file in /tmp directory and check its md5 sum."):
+        test_file = fs_utils.create_test_file(f'{test_file_path}',
+                                              'There is nothing interesting inside this file.')
```

Comment:
It's so unsatisfying that we don't have in TF elegant method of generating reasonable* test files yet.
In this test this test file is practically meaningless and creating it could be removed without any loss, but in general this looks quite sad.

\* By reasonable I mean bigger than one sector containing joke string instead of verifiable pattern.

Reply:
I've changed this function in TF. I think its more reasonable now, but please, check it - https://github.com/Open-CAS/test-framework/pull/114.